src/views/virtualmachines/details/tabs/details/hooks/useSSHService.tsx

```diff
@@ -1,15 +1,39 @@
+import * as React from 'react';
+
 import { ServiceModel } from '@kubevirt-ui/kubevirt-api/console';
 import { IoK8sApiCoreV1Service } from '@kubevirt-ui/kubevirt-api/kubernetes';
 import { V1VirtualMachineInstance } from '@kubevirt-ui/kubevirt-api/kubevirt';
 import { useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
 
-const useSSHService = (vmi: V1VirtualMachineInstance) => {
-  return useK8sWatchResource<IoK8sApiCoreV1Service>({
+export type UseSSHServiceReturnType = [service: IoK8sApiCoreV1Service, loaded: boolean];
+
+const useSSHService = (vmi: V1VirtualMachineInstance): UseSSHServiceReturnType => {
+  const [sshService, setSSHService] = React.useState<IoK8sApiCoreV1Service>();
+  const [loaded, setLoaded] = React.useState(false);
+
+  const [services, servicesLoaded, servicesError] = useK8sWatchResource<IoK8sApiCoreV1Service[]>({
     kind: ServiceModel.kind,
-    isList: false,
+    isList: true,
     namespace: vmi?.metadata?.namespace,
-    name: `${vmi?.metadata?.name}-ssh-service`,
   });
+
+  React.useEffect(() => {
+    if (servicesLoaded && !sshService && vmi) {
+      const vmiServices = services.filter(
+        (service) => service?.spec?.selector?.['vm.kubevirt.io/name'] === vmi?.metadata?.name,
+      );
+
+      const sshVMIService = vmiServices.find((service) =>
+        service?.spec?.ports?.find((port) => parseInt(port.targetPort, 10) === 22),
```

Comment:
```suggestion
        service?.spec?.ports?.find((port) => parseInt(port.targetPort, 10) === SSH_PORT),
```
22 is a magic number, best to give it a name with semantics